packages/kbn-alerts-ui-shared/src/common/hooks/use_alerts_data_view.ts

```diff
@@ -35,7 +35,13 @@ export interface UseAlertsDataViewParams {
 
 export interface UseAlertsDataViewResult {
   isLoading: boolean;
-  dataView?: Omit<DataViewBase, 'fields'> & { fields: FieldSpec[] };
+  dataView?: Omit<DataViewBase, 'fields'> & {
+    fields: FieldSpec[];
+    /**
+     * Indicates if the end user has read access to the index, besides the alerting authorization
+     */
+    hasReadIndexPrivilege?: boolean;
```

Comment:
question about this field, shouldn't we do this implicitly on the backend based on the request user? Seems a bit odd to expose a field like this to the client.

Reply:
The main use case is authorizing views like the filter controls bar that are unaware of the alerting RBAC and directly work with the indices. In these cases we don't have a way to know if the end user (besides their alerting permissions) has direct read access to the alerting indices. Since this is particularly useful for unprivileged users, checking the index permissions directly on the client seems a bit fragile: the user should have access to the plugin(s) that expose the privilege check functionality, which is unlikely in this case.
I could be missing a more straightforward way to check this so please if you have any ideas let me know 🙂

Reply:
Right but this means anyone could "spoof" the read index privilege by passing this field? At that point you might as well just always assume the user has privilege

Reply:
> Right but this means anyone could "spoof" the read index privilege by passing this field? At that point you might as well just always assume the user has privilege

Not sure I follow you here. This is set by the server, not the client, (I assume) similarly to how the `capabilities` object informs the client on what the user has access to for example.

> Since we have the request, we'll have the user, can't we check the user's index privilege? unless we're someone making a request on behalf of another user or something.

This privilege is not instrumental to authenticating the `alerts/_index` request itsefl, but to tell the client if the user has access to the ES alerts index itself. Unprivileged users might not have a direct access but still be able to indirectly access them through the alerting APIs.

The controls embeddable thought works directly with the ES indices, not through the alerting APIs. That's why we need to check if the user has "direct" access to those indices: to hide the controls and avoid 403 errors.